EventFilter/HcalRawToDigi/src/HcalDataFrameFilter.cc

```diff
@@ -92,6 +129,29 @@ ZDCDigiCollection HcalDataFrameFilter::filter(const ZDCDigiCollection& incol, Hc
   return output;
 }
 
+QIE10DigiCollection HcalDataFrameFilter::filter(const QIE10DigiCollection& incol, HcalUnpackerReport& r) {
+  QIE10DigiCollection output(incol.samples());
+  for (QIE10DigiCollection::const_iterator i=incol.begin(); i!=incol.end(); i++) {
+    if (!HcalDataFrameFilter_impl::checkQIE10(*i,requireCapid_,requireDVER_))
+      r.countBadQualityDigi(i->id());
+    // Never exclude QIE10 digis as their absence would be
+    // treated as a digi with zero charged deposited in that channel
+    output.push_back(*i);
```

Comment:
Yes, Kevin is correct here. The difference with other dataframes is that, for QIE10, two dataframes are combined to make one rechit. Each dataframe carries the data from one PMT anode, while the rechit represents the whole PMT with two anodes. If one anode dataframe is corrupt while the other is not, the proper way to handle this situation is to multiply the energy seen by the "good" anode by two. On the other hand, if the anode dataframe is simply absent, the most probable reason for that is zero suppression, in which case the energy of the anode present in the data should stay unmodified.